### PR TITLE
grid summary widget  created 

### PR DESCRIPTION
**Pull request by cs-pranali-chavan**
grid summary widget  created to be embedded to show grid data in fortiguard IOC search or any other widget through external json

### CODE REVIEW

File: widget/edit.controller.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        .module('cybersponse')
+        .controller('editGridSummary100Ctrl', editGridSummary100Ctrl);
+
+    editGridSummary100Ctrl.$inject = ['$scope', '$uibModalInstance', 'config', 'widgetUtilityService', '$timeout'];
+
+    function editGridSummary100Ctrl($scope, $uibModalInstance, config, widgetUtilityService, $timeout) {
+        $scope.cancel = cancel;
+        $scope.save = save;
+        $scope.config = config;
+        $scope.header = $scope.config.title ? 'Edit Grid Summary' : 'Add Grid Summary';
```
Comment (cs-suyog-jinturkar) [inline widget/edit.controller.js:17]: i18n supported title?

File: widget/edit.controller.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        $scope.setGridDefinition = setGridDefinition; //set grid options
+        $scope.setGridDataJson = setGridDataJson; //set grid data
+        $scope.config.gridDefinitionJson = !angular.isArray($scope.config.gridDefinitionJson) ? $scope.config.gridDefinitionJson : {};
+        $scope.jsoneditorOptions = {
+          name: 'Fields',
+          mode: 'code',
+          onEditable: function () {
+            return {
+              field: true,
```
Comment (cs-suyog-jinturkar) [inline widget/edit.controller.js:26]: repeated code alert, can we optimise this by parameterizing it?

File: widget/edit.controller.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            });
+          }
+        }
+
+        function setGridDefinition(json) {
+          if (angular.isString(json)) {
+            try {
+              $scope.config.gridDefinitionJson = JSON.parse(json);
+            } catch (e) {
+              // invalid JSON. skip the rest
```
Comment (cs-suyog-jinturkar) [inline widget/edit.controller.js:54]: any console error to be shown here?

File: widget/info.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+            "Reports",
+            "Listing"
+        ],
+        "certified": "Yes",
+        "publisher": "Fortinet",
+        "compatibility": [
+            "7.6.1"
+        ],
+        "category": [
+            "Threat Intelligence"
```
Comment (cs-suyog-jinturkar) [inline widget/info.json]: category can be other the TI, right?

File: widget/info.json [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        "description": "",
+        "pages": [
+            "View Panel",
+            "Dashboard",
+            "Reports",
+            "Listing"
+        ],
+        "certified": "Yes",
+        "publisher": "Fortinet",
+        "compatibility": [
```
Comment (cs-suyog-jinturkar) [inline widget/info.json:18]: any reason for compatibility 761?

File: widget/info.json [hunk 6]
```diff
@@ -0,0 +1,26 @@
+{
+    "name": "gridSummary",
+    "title": "Grid Summary",
+    "subTitle": "create grid for given json data",
```
Comment (cs-suyog-jinturkar) [inline widget/info.json]: Create a grid and populate given JSON data in it

File: widget/info.json [hunk 7]
```diff
@@ -0,0 +1,26 @@
+{
+    "name": "gridSummary",
+    "title": "Grid Summary",
```
Comment (cs-suyog-jinturkar) [inline widget/info.json]: we might need to change the title/name to something like Summary Grid

File: widget/widgetAssets/js/widgetUtility.service.js [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+    Copyright (c) 2024 Fortinet Inc
+Copyright end */
+'use strict';
+
+(function () {
+  angular
+    .module('cybersponse')
+    .factory('widgetUtilityService', widgetUtilityService);
+
+  widgetUtilityService.$inject = ['$q', '$http', '$injector', '$interpolate', 'toaster'];
```
Comment (cs-suyog-jinturkar) [inline widget/widgetAssets/js/widgetUtility.service.js:12]: suggestion: this is a common code and can be a part of widgetUtilities widget and all widget can refer it

File: widget/edit.controller.js (hunk 3 above)
Comment (cs-pranali-chavan) [reply widget/edit.controller.js:54]: not required

File: widget/info.json (hunk 4 above)
Comment (cs-pranali-chavan) [reply widget/info.json]: yes

File: widget/info.json (hunk 5 above)
Comment (cs-pranali-chavan) [reply widget/info.json:18]: nopes .. added as per parallel release

File: widget/info.json (hunk 7 above)
Comment (cs-pranali-chavan) [reply widget/info.json]: name will be same as per the widget repo 
title we can change

File: widget/edit.controller.js (hunk 1 above)
Comment (cs-pranali-chavan) [reply widget/edit.controller.js:17]: yes will add i18n changes

File: widget/info.json [hunk 9]
```diff
@@ -1,7 +1,7 @@
 {
     "name": "gridSummary",
-    "title": "Grid Summary",
-    "subTitle": "create grid for given json data",
+    "title": "Summary grid",
```
Comment (cs-suyog-jinturkar) [inline widget/info.json]: Grid

**Pull request merged**